src/main/java/sg/dex/starfish/impl/file/FileAsset.java

```diff
@@ -23,96 +21,135 @@
-	 * @return FileAsset
-	 */
-	public static FileAsset create(File f) {
-		String metaString = JSON.toString(buildMetadata(f,null));
-		return new FileAsset(metaString, f);
-	}
-
-	/**
-	 * Create a FileAsset to read from an existing file
-	 *
-	 * @param f file handle one which the File Asset will be created
-	 * @return FileAsset
-	 */
-	public static FileAsset create(File f,String metaData) {
-		String metaString = JSON.toString(buildMetadata(f,metaData));
-		return new FileAsset(metaString, f);
-	}
-
-	/**
-	 * Build default metadata for a file asset
-	 *
-	 * @param f The file to use for this file asset
-	 * @return The default metadata as a String
-	 */
-	protected static Map<String, Object> buildMetadata(File f,String meta) {
-		try {
-
-			String content = Utils.stringFromStream(new FileInputStream(f));
-			String hashOfContent = Hex.toString(Hash.keccak256(content));
-
-			Map<String, Object> ob = Utils.createDefaultMetadata();
-			ob.put(TYPE, DATA_SET);
-			ob.put(SIZE, f.length());
-			ob.put(FILE_NAME, f.getName());
-			ob.put(CONTENT_TYPE, OCTET_STREAM);
-			ob.put(CONTENT_HASH, hashOfContent);
-
-			if (meta != null) {
-				Map<String,Object> metaMap = JSON.toMap(meta);
-				for (Map.Entry<String, Object> me : metaMap.entrySet()) {
-					ob.put(me.getKey(), me.getValue());
-				}
-			}
-
-			return ob;
-
-		} catch (FileNotFoundException e) {
-			throw new StarfishValidationException(e.getMessage());
-		}
-	}
-
-	/**
-	 * Gets an input stream that can be used to consume the content of this asset.
-	 *
-	 * Will throw an exception if consumption of the asset data in not possible
-	 * locally.
-	 * 
-	 * @throws AuthorizationException if requestor does not have access permission
-	 * @throws StorageException if unable to load the Asset
-	 * @return An input stream allowing consumption of the asset data
-	 */
-	@Override
-	public InputStream getContentStream() {
-		try {
-			return new FileInputStream(file);
-		}
-		catch (FileNotFoundException e) {
-			throw new StorageException("File not found ,file : "+file , e);
-		}
-	}
-
-	@Override
-	public long getContentSize() {
-		return null != file ? file.length(): -1;
-	}
-
+    private final File file;
+
+    protected FileAsset(String meta, File file) {
+        super(meta);
+        this.file = file;
+    }
+
+    /**
+     * This method is to crete Resource Asset with specific resource path, metadata  and isHashOfContentRequired
+     *
+     * @param f                       path of the resource
+     * @param isHashOfContentRequired if true then hash of content is calculated and included in metadata.
+     *                                Hash of content will be calculated using Keccak256 hash function
+     *                                if false then content hash is not included in metadata
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f, boolean isHashOfContentRequired) {
```

Comment:
The isContentHashRequired should be optional. 

I suggest a separate method in Asset: Asset.includeContentHash() which takes any asset and computes / adds a content hash as required.